src/analytics/redux/tracked_events.js

```diff
@@ -19,6 +69,66 @@ function getBlockObject( block ) {
 	return select( 'core/block-editor' ).getBlock( block ) || {};
 }
 
+/**
+ * Track block replacement.
+ *
+ * @param {Array}          originalBlockIds ID(s) or blocks that are being replaced
+ * @param {Object | Array} blocks           Block instance object or an array of such objects
+ * @param {number}         indexToSelect    Index of replacement block to select.
+ * @param {0|-1|null}      initialPosition  Index of caret after in the selected block after the operation.
+ * @param {?Object}        meta             Optional Meta values to be passed to the action object.
+ *
+ * @return {void}
+ */
+const trackBlockReplacement = (
+	originalBlockIds,
+	blocks,
+	indexToSelect,
+	// eslint-disable-next-line no-unused-vars
+	initialPosition = 0,
+	meta = {}
+) => {
+	const insert_method = getBlockInserterUsed( originalBlockIds, meta );
+
+	// To avoid tracking block insertions when replacing a block, we only track replacements
+	// when the slash inserter is used.
+	if ( insert_method === INSERTERS.SLASH_INSERTER ) {
+		trackBlocksHandler( blocks, 'editor_block_inserted', ( { name } ) => ( {
+			block_name: name,
+			insert_method,
+		} ) );
+	}
+};
+
+/**
+ * Track block insertion.
+ *
+ * @param {Object | Array} blocks          Block instance object or an array of such objects
+ * @param {?number}        index           Index at which block should be inserted.
+ * @param {?string}        rootClientId    Optional root client ID of block list on which to insert.
+ * @param {?boolean}       updateSelection If true block selection will be updated. If false, block selection will not change. Defaults to true.
+ * @param {0|-1|null}      initialPosition Initial focus position. Setting it to null prevent focusing the inserted block.
+ * @param {?Object}        meta            Optional Meta values to be passed to the action object.
+ *
+ * @return {void}
+ */
+function trackBlockInsertion(
+	blocks,
+	index,
+	rootClientId,
+	updateSelection,
+	// eslint-disable-next-line no-unused-vars
+	initialPosition = 0,
```

Comment:
I think we don't need to disable the ESLint rule if we don't set a default value for `initialPosition`. We are not using this argument, so I'd say it's safe to remove the default value.

Reply:
Thanks for the suggestion! I ended up removing this one in https://github.com/wordpress-mobile/gutenberg-mobile/pull/5976/commits/94df0273bad2f0dc1a9422b4a590e5ecb7b3beb5#diff-b264ddd07fecd2f9655557832fb8f89baa62f7f89c387e80df664c26ceb36084L121 as your other suggestion about unneeded params, thanks!